ee/tabby-webserver/src/service/auth.rs

```diff
@@ -420,14 +421,17 @@ impl AuthenticationService for AuthenticationServiceImpl {
         provider: OAuthProvider,
     ) -> std::result::Result<OAuthResponse, OAuthError> {
         let client = oauth::new_oauth_client(provider, Arc::new(self.clone()));
-        let email = client.fetch_user_email(code).await?;
+        let access_token = client.exchange_code_for_token(code).await?;
+        let email = client.fetch_user_email(&access_token).await?;
+        let name = client.fetch_user_full_name(&access_token).await?;
         let license = self
             .license
             .read()
             .await
             .context("Failed to read license info")?;
         let user_id =
-            get_or_create_oauth_user(&license, &self.db, &self.setting, &self.mail, &email).await?;
+            get_or_create_oauth_user(&license, &self.db, &self.setting, &self.mail, &email, &name)
```

Comment:
we fill in user name from this method, so I improved unit test for this method to cover the change here.